### PR TITLE
New blip color, Set vehicle door angle, Get current vehicle weapon Hash

### PR DESCRIPTION
**Pull request by qiangqiang101**

### CODE REVIEW

**Comment by crosire** [comment]
You do not have to create a new pull request when you made a mistake. Just push a new commit to the same branch to fix it and the pull request is updated automatically ;)

**Comment by qiangqiang101** [comment]
@crosire noted

File: source/scripting/Vehicle.cs [hunk 1]
```diff
@@ -1429,6 +1429,12 @@ public int PassengerCount
 				return Function.Call<int>(Hash.GET_VEHICLE_NUMBER_OF_PASSENGERS, Handle);
 			}
 		}
+        public int GetVehicleWeapon(Ped Ped)
```
Comment (crosire) [inline source/scripting/Vehicle.cs]: Could this return a specific type other than `int`? Is it a `WeaponHash` perhaps, or does `GET_CURRENT_PED_VEHICLE_WEAPON` give back a handle to a weapon instance?

Comment (qiangqiang101) [inline source/scripting/Vehicle.cs]: I use it like this

```
                If GetVehicleWeapon(Game.Player.Character) = [Function].Call(Of Integer)(Hash.GET_HASH_KEY, "VEHICLE_WEAPON_PLAYER_LAZER") Then
                        Game.Player.Character.CurrentVehicle.OpenDoor(VehicleDoor.Trunk, False, False)
                End If
```

**Comment by qiangqiang101** [comment]
@crosire I use it like this

```
                If GetVehicleWeapon(Game.Player.Character) = [Function].Call(Of Integer)(Hash.GET_HASH_KEY, "VEHICLE_WEAPON_PLAYER_LAZER") Then
                        Game.Player.Character.CurrentVehicle.OpenDoor(VehicleDoor.Trunk, False, False)
                End If
```

File: source/scripting/Vehicle.cs (hunk 1 above)
Comment (zorg93) [inline source/scripting/Vehicle.cs]: I think a new enum should be made for VehicleWeaponHash (they are defined in the same files as Weapons) We dont want users mixing WeaponHash and VehicleWeaponHash when say giving a ped a weapon. IIRC there is also a way to set the VehicleWeapon

**Comment by crosire** [comment]
Still waiting for the return type of `GetVehicleWeapon` to be changed to an enum. Will have to close this otherwise.

**Comment by Fireboyd78** [comment]
@crosire you could always accept his pull request and add it afterwards.

**Comment by Fireboyd78** [comment]
@qiangqiang101 You need to convert spaces to tabs.

**Comment by zorg93** [comment]
Due to lack of activity I've closed this, the changes added(plus the fixes needed) have been added in here https://github.com/crosire/scripthookvdotnet/pull/541/commits/39e93fc4d040b504327c9168ae980179c7d040df

**Pull request closed without merging**